app/github-login-button.tsx

```diff
@@ -0,0 +1,57 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import useSWR from 'swr';
+
+const fetcher = (url: string) => fetch(url).then((res) => res.json());
+
+const GitHubLoginButton: React.FC = () => {
+  const [token, setToken] = useState<string | null>();
+  const { data, error, isLoading } = useSWR(token ? `/api/github/user/${token}` : null, fetcher);
```

Comment:
Ditto, escape token for defense in depth. Here, use [`encodeURI`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI) since it's not a query component.